library/ssl_misc.h

```diff
@@ -546,14 +517,14 @@ struct mbedtls_ssl_handshake_params
      *   renegotiating or resuming a session, it is equal to the previously
      *   negotiated minor version.
      *
-     *   There is no maximum minor version field in this handshake context.
+     *   There is no maximum TLS version field in this handshake context.
      *   From the start of the handshake, we need to define a current protocol
-     *   version for the record layer which we define as the maximum minor
-     *   version to be negotiated. The `minor_ver` field of the SSL context is
+     *   version for the record layer which we define as the maximum TLS
+     *   version to be negotiated. The `tls_version` field of the SSL context is
      *   used to store this maximum value until it contains the actual
      *   negotiated value.
      */
-    unsigned char min_minor_ver;
+    uint16_t min_tls_version;
```

Comment:
```suggestion
    mbedtls_ssl_protocol_version min_tls_version;
```

Reply:
Given default alignment, this suggested change will grow the structure +4 bytes.  If we change where the member is, we can reduce that to +2 bytes.  Ok if I move the member below the two uint8_t members which currently follow it?

Reply:
:rocket: implemented as (signed bitfield) `mbedtls_ssl_protocol_version MBEDTLS_PRIVATE(min_tls_version) : 16;`

Reply:
:rocket: reverted the signed bitfield due to compatibilty, but left conversion to `mbedtls_ssl_protocol_version`

Given the preprocessor conditionals that follow, should I move the `min_tls_version` member below the following two uint8_t elements to possibly reduced (default) structure padding?